contracts/VolOracle.sol

```diff
@@ -129,9 +141,9 @@ contract VolOracle {
                 tickSquareDelta + prevObservation.tickSquareCumulative
             );
         }
-
+        (uint32 lastBlockTimestamp, , , ) = uniPool.observations(endIndex % poolCardinality);
         volOracleState.observationIndex = volObservationIndex;
         volOracleState.lastObservationIndex = endIndex % poolCardinality;
-        volOracleState.lastBlockTimestamp = block.timestamp;
+        volOracleState.lastBlockTimestamp = uint256(lastBlockTimestamp);
```

Comment:
we also change the lastBlockTimestamp value when we initialize the pool. To keep it consistent, we'd better update there as well